near-sdk/src/types/error.rs

```diff
@@ -32,3 +32,30 @@ where
         crate::env::panic_str(self.as_ref())
     }
 }
+
+/// A simple type used in conjunction with [FunctionError] representing that the function should
+/// abort without a custom message.
+///
+/// ```
+/// use near_sdk::borsh::{BorshDeserialize, BorshSerialize};
+/// use near_sdk::{Abort, near_bindgen};
+///
+/// #[near_bindgen]
+/// #[derive(Default, BorshDeserialize, BorshSerialize)]
+/// pub struct Contract;
+///
+/// #[near_bindgen]
+/// impl Contract {
+///     #[handle_result]
+///     pub fn foo(&self) -> Result<Result<String, &'static str>, Abort> {
```

Comment:
Opinionated but seems more worth using the example with an actual pattern, `Abort` wouldn't have to be used for nested `Result` and this might be confusing

Reply:
Hmm, I am confused. Isn't this the exact pattern we are trying to promote with `Abort`?

Reply:
Well, it makes this pattern easier, but this abort can be used more generally. If someone just wanted a minimal way to abort when errors were hit to avoid debug messages, this could be useful.